src/commands/apm_cmds/publish.js

```diff
@@ -208,8 +211,30 @@ exports.task = function ({
       title: 'Preflight checks for publishing to APM',
       enabled: () => !automaticallyBump,
       task: (ctx) => new TaskList([
+        {
+          title: 'Fetch latest app version from chain and increment',
+          enabled: () => !!versionBump,
+          task: async (ctx, task) => {
+
+            if (VERSION_BUMP_TYPES.indexOf(versionBump) == -1) {
+              throw new Error(`Not a valid version bump type. Please use one of 'patch', 'minor', 'major'`)
+            }
+
+            const newVersion = await versionTask.bumpVersionAndUpdateManifest({
```

Comment:
This API now always resolves with a version either way. Please review this carefully.